hyperlane/cmd/hyp/cmd/events.go

```diff
@@ -1,19 +1,40 @@
 package cmd
 
 import (
+	"fmt"
 	"log"
 
 	"github.com/bcp-innovations/hyperlane-cosmos/util"
 	ismtypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/01_interchain_security/types"
 	hooktypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/02_post_dispatch/types"
 	coretypes "github.com/bcp-innovations/hyperlane-cosmos/x/core/types"
 	warptypes "github.com/bcp-innovations/hyperlane-cosmos/x/warp/types"
+	zkismtypes "github.com/celestiaorg/celestia-app/v6/x/zkism/types"
 	abci "github.com/cometbft/cometbft/abci/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/gogoproto/proto"
 )
 
-func parseISMFromEvents(events []abci.Event) util.HexAddress {
+func parseIsmIDFromZkISMEvents(events []abci.Event) util.HexAddress {
+	var ismID util.HexAddress
+	for _, evt := range events {
+		if evt.GetType() == proto.MessageName(&zkismtypes.EventCreateZKExecutionISM{}) {
+			event, err := sdk.ParseTypedEvent(evt)
+			if err != nil {
+				log.Fatalf("failed to parse typed event: %v", err)
+			}
+
+			if ismEvent, ok := event.(*zkismtypes.EventCreateZKExecutionISM); ok {
```

Comment:
what about the branch of `!ok`?

Reply:
then you've probably invoked the wrong tx rpc and will get an empty ism id returned. 

This code is not for production and is basically deployment scripts for our testing environment.